src/hotspot/share/runtime/serviceThread.cpp

```diff
@@ -126,8 +126,9 @@ void ServiceThread::service_thread_entry(JavaThread* jt, TRAPS) {
               (cldg_cleanup_work = ClassLoaderDataGraph::should_clean_metaspaces_and_reset()) |
               (jvmti_tagmap_work = JvmtiTagMap::has_object_free_events_and_reset())
              ) == 0) {
-        // Wait until notified that there is some work to do.
-        ml.wait();
+        // Wait until notified that there is some work to do or timer expires.
+        // OopStorage work needs to be done at periodic intervals.
```

Comment:
Rather than calling out OopStorage here, maybe just say some cleanup requests don't notify the
ServiceThread, instead relying on it to run periodically.  After this change we might want to audit
other cleanup requests and decide if they actually need to notify the ServiceThread in order to get a
more prompt response, or could just wait for the next periodic wakeup.

Reply:
I reworded the comment.  Some might make sense on a periodic timer. Not sure about others.  The ones that trigger more are the StringTable and SymbolTable.